### PR TITLE
feat: bring rich text validator [TOL-2426]

### PR DESCRIPTION
**Pull request by YvesRijckaert**
- bring rich text validator
- migrate tests
- removed JSON schema and generation
- use it later in the backend and in field editors cypress tests

BREAKING CHANGE:
removed `getSchemaWithNodeType` in favor of `validateRichTextDocument`
no longer bundle dependencies

### CODE REVIEW

File: packages/rich-text-types/src/validator/assert.ts [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+        return false;
+      }
+    }
+
+    if (isPlainObject(value)) {
+      return true;
+    }
+
+    const path = key ? this.path.of(key) : this.path;
+    const property = key ?? this.path.last() ?? 'value';
```
Comment (z0al) [inline packages/rich-text-types/src/validator/assert.ts:75]:
```suggestion
    const property = key ? key : 'value';
```

File: packages/rich-text-types/src/validator/index.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  [BLOCKS.TABLE_ROW]: assert(() => ({
+    nodeTypes: [BLOCKS.TABLE_CELL, BLOCKS.TABLE_HEADER_CELL],
+    min: 1,
+  })),
+  [BLOCKS.TABLE_CELL]: assertTableCell,
+  [BLOCKS.TABLE_HEADER_CELL]: assertTableCell,
+  [INLINES.HYPERLINK]: new HyperLinkAssertion(),
+  [INLINES.EMBEDDED_ENTRY]: assertVoidEntryLink,
+  [INLINES.EMBEDDED_RESOURCE]: assertLink('Contentful:Entry', VOID_CONTENT),
+  [INLINES.ENTRY_HYPERLINK]: assertLink('Entry', ['§']),
```
Comment (z0al) [inline packages/rich-text-types/src/validator/index.ts:74]: what is this `§`??

**Review by z0al — COMMENTED** [review]
LGTM. just one question

File: packages/rich-text-types/src/validator/index.ts (hunk 2 above)
Comment (z0al) [reply packages/rich-text-types/src/validator/index.ts:74]: @YvesRijckaert / @chrishelgert ☝🏻

Comment (chrishelgert) [reply packages/rich-text-types/src/validator/index.ts:74]: let me check 👁️

Comment (z0al) [reply packages/rich-text-types/src/validator/index.ts:74]: It's supposed to be `text` just like the other links I think. 

But the fact this didn't fail any test is a bit alarming 🙈 . Could you add some bitte 🙇🏻 ?

Comment (chrishelgert) [reply packages/rich-text-types/src/validator/index.ts]: typo from copying over, will extend the test to cover this

Comment (chrishelgert) [reply packages/rich-text-types/src/validator/index.ts]: test extended 👍

**Comment by z0al** [comment]
The bundle size results seem rather strange, worth double checking @YvesRijckaert / @chrishelgert

**Comment by chrishelgert** [comment]
> The bundle size results seem rather strange, worth double checking @YvesRijckaert / @chrishelgert

One thing is lodash, I removed that locally already - we still have some increase will double check that too

File: packages/rich-text-types/rollup.config.js [hunk 3]
```diff
@@ -1,32 +1,6 @@
...
-  const baseConfig = config(outputFile);
-
-  return {
-    ...baseConfig,
-    output: {
-      // Maintain the same file names & folders structure as "src"
-      // This is necessary to resolve generated JSON schema files
-      //
-      // https://rollupjs.org/guide/en/#outputpreservemodules
-      preserveModules: true,
```
Comment (chrishelgert) [inline packages/rich-text-types/rollup.config.js:16]: @z0al that here

**Review by z0al — APPROVED** [review]

**Pull request merged**